contrib/boost/CMakeLists.txt

```diff
@@ -24,6 +24,19 @@ file(GLOB_RECURSE NOT_BOOST_SOURCES
 
 list(REMOVE_ITEM BOOST_SOURCES ${NOT_BOOST_SOURCES})
 
+# Apply patch for Intel Compiler; http://stackoverflow.com/a/41188485
+if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
```

Comment:
No, the same error (but with different file) is in 1.63. In 1.64 there is a bug that prevents from compiling boost::mpi (sic!); 
1.60 did not compile with clang so 1.61 was the first 1.6x version that run on travis. I must admit that it is a nightmare.